R/SIRD_plot.R

```diff
@@ -1,121 +1,158 @@
-          )
-      })
-    })
+        )
+}
+
 
-    output$SIRDPhasePlane <- renderPlot({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        ggplot(val, aes(x = S)) +
-          geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
-          theme(
-            axis.line = element_line(color = "black"), axis.text = element_text(size = 14),
+plotPhasePlaneSIRD <- function() {
+    valuesSIRD <- solveSIRD()
+    ggplot(valuesSIRD, aes(x = S)) +
+        geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
+        theme(
+            axis.line = element_line(color = "black"),
+            axis.text = element_text(size = 14),
             axis.title.x = element_text(size = 16, face = "bold"),
             axis.title.y = element_text(size = 16, face = "bold")
-          ) +
-          theme(legend.position = "bottom") +
-          ggtitle("SI Phase Plane") +
-          theme(plot.title = element_text(size = 22, face = "bold")) +
-          ylab("Infected (I)") +
-          scale_x_continuous(expand = c(0, 0)) +
-          scale_y_continuous(expand = c(0, 0)) +
-          xlab("Susceptible (S)") +
-          scale_color_identity(
-            breaks = "Blue",
-            labels = "Susceptible"
-          )
-      })
-    })
+        ) +
+        theme(legend.position = "bottom") +
+        ggtitle("SI Phase Plane") +
+        theme(plot.title = element_text(size = 22, face = "bold")) +
+        ylab("Infected (I)") +
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        xlab("Susceptible (S)") +
+        scale_color_identity(
+            name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
+            labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
+        )
+}
 
-    output$tableSIRD <- renderTable({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        val <- val[-c(7)]
-        return(val)
-      })
+# no longer need to do shiny::isRunning() check since this function is only called
+# from server.R
+solveAndRenderSIRD <- function() {
+    expr <- quote({
+        output$plotSIRD <- renderPlot({
+            plotSIRD()
+        })
+        output$SIRDPhasePlane <- renderPlot({
+            plotPhasePlaneSIRD()
+        })
+        output$tableSIRD <- renderTable({
+            return(solveSIRD()[-c(6)])
+        })
     })
-  })
 
-  ## Evaluate the above qouted expression in the parent frame (the calling
-  ## environment).
-  eval.parent(expr)
+    eval.parent(expr)
 }
+
+# For standalone usage
+# TODO: find out if there's a better way to check if the script is being 
+# run standalone or as part of a shiny app
+if (!shiny::isRunning()) {
+    plot <- plotSIRD()
+    print(plot)
+    ggsave("sird_plot.png", plot)
+
+    phasePlanePlot <- plotPhasePlaneSIRD()
+    print(phasePlanePlot)
+    ggsave("sird_phase_plane_plot.png", phasePlanePlot)
+}
```

Comment:
While I respect what you're doing here, during an interactive R session it should be left up to the user whether they save a plot or other object to disk. Due to that user choice, this entire section can be removed. @ashokkrish's request to be able to use these files and functions interactively in a workshop will be served well with a basic R package that he may write at a later date (which would include the _example function calls_ in package documentation). For our purposes, this section is unnecessary.

In Python, there is an expectation that every module may be called directly as `__main__`, and thus it should "do" something. In R there is no expectation to source modules and have them accomplish anything other than loading the definitions written in the file because the R session is interactive (usually). When R is used in a batch mode things like this may be done, but that is not our use case here.

To answer your question, however, you're very close to a bulletproof solution. Here is the minor change that would enable the check you want to perform.
```r
if ("shiny" %in% installed.packages() && !shiny::isRunning()) { ... }
```

Reply:
Thanks for still showing me how to check it correctly.